<?xml version="1.0"?>
<launch>
  <arg name="sim" default="false"/>
  <arg name="viz" default="false"/>
  <arg name="robot_type" default="wolfgang"/>

  <let if="$(env IS_ROBOT false)" name="taskset" value="taskset -c 3,4"/>
  <let unless="$(env IS_ROBOT false)" name="taskset" value=""/>

  <include file="$(find-pkg-share bitbots_parameter_blackboard)/launch/parameter_blackboard.launch" >
      <arg name="sim" value="$(var sim)"/>
  </include>

  <include file="$(find-pkg-share bitbots_robot_description)/launch/load_robot_description.launch" >
      <arg name="sim" value="$(var sim)"/>
    <arg name="robot_type" value="$(var robot_type)"/>
  </include>

  <!-- launch the base footprint -->
  <node pkg="humanoid_base_footprint" exec="base_footprint" output="screen">
    <param name="support_state_topics" value="[walk_support_state, dynamic_kick_support_state]"/>
    <param name="use_sim_time" value="$(var sim)"/>
  </node>

  <!-- launch the odometry -->
  <include file="$(find-pkg-share bitbots_odometry)/launch/odometry.launch">
    <arg name="sim" value="$(var sim)"/>
  </include>

  <group unless="$(var sim)">
    <group unless="$(var viz)">
      <include file="$(find-pkg-share bitbots_ros_control)/launch/ros_control.launch" />
      <node pkg="bitbots_quintic_walk" exec="WalkNode" output="screen" launch-prefix="$(var taskset)">
        <param from="$(find-pkg-share bitbots_quintic_walk)/config/walking_$(var robot_type)_robot.yaml"/>
        <remap from="/walking_motor_goals" to="/DynamixelController/command"/>
      </node>
    </group>
  </group>

  <group if="$(var sim)">
    <node pkg="bitbots_quintic_walk" exec="WalkNode" output="screen" launch-prefix="$(var taskset)">
      <remap from="/walking_motor_goals" to="/DynamixelController/command"/>
      <param from="$(find-pkg-share bitbots_quintic_walk)/config/walking_$(var robot_type)_simulator.yaml"/>
    </node>
  </group>

  <group if="$(var viz)">
    <node pkg="bitbots_quintic_walk" exec="WalkNode" output="screen">
        <param from="$(find-pkg-share bitbots_quintic_walk)/config/walking_$(var robot_type)_viz.yaml"/>
    </node>
    <node pkg="bitbots_utils" exec="motor_goals_viz_helper.py" output="screen" args="--walking --robot-type $(var robot_type)"/>
    <node pkg="rviz2" exec="rviz2" args="-d $(find-pkg-share bitbots_quintic_walk)/config/walk.rviz"/>
    <node pkg="bitbots_utils" exec="dummy_imu.py"/>
  </group>

</launch>
